### ISSUE TITLE
Investigate Pipeline Caching to improve build performance.

### ISSUE THREAD

**Issue by mitchdenny** [Azure/azure-sdk#496]
Pipeline Caching, a feature of Azure Pipelines is rolling out and we should investigate adopting it to see if it can reduce our CI times, particular around dependency restoration (but we should look at all options).

**Comment by mitchdenny** [comment]
The Azure Artifacts team has started using tar to bundle small loose files into a larger archive which will improve download times for things like NPM packages (lots of small files). Given this its probably going to be worth experimenting again in the JavaScript space to see if we can any wins.

One added bonus is that with the tar archive symbolic links are preserved which means it will hopefully be rush friendly.

**Comment by mitchdenny** [comment]
@KarishmaGhiya ... did you see much improvement on Node.js around caching performance?

**Comment by mitchdenny** [comment]
Closing this down. It helped for Java ... marginally (switching to our own Maven proxy was a bigger impact). For JS I don't think its going to help because of Rush and the way it symlinks directories. For .NET it won't make much difference because of the fairly small dependency set (and the improved performance of NuGet lately).

Perhaps the biggest change we made from when this was first created was the cut over to unified pipelines which really narrowed the scope of what we were restoring in the first place.
